sound/soc/intel/boards/sof_sdw_max98373.c

```diff
@@ -55,6 +56,13 @@ static int spk_init(struct snd_soc_pcm_runtime *rtd)
 	if (ret)
 		dev_err(rtd->dev, "failed to add first SPK map: %d\n", ret);
 
+	/*
+	 * enable two channels and use mixer control 'DAI Sel Mux'
+	 * to select channel for each codec
+	 */
+	for_each_rtd_codec_dais(rtd, i, codec_dai)
+		snd_soc_dai_set_tdm_slot(codec_dai, 0x3, 0x3, 2, 32);
```

Comment:
Hang on, there could be a problem here.
The idea of using TDM slots is that when the slot configuration is the same, all amplifiers would receive the same data from the same slots and extract it 
For capture, you cannot use the same slots for the two amplifiers, that would lead to a bus clash.
Can you clarify @RanderWang what setting the same TX mask for the amplifiers? How does this impact the bit allocation for the Source port of the amplifiers?

Reply:
@plbossart a dumb question here. The other PR that was recently approved had similar TDM slot settings too. Is the problem you're point out SDW specific?

Reply:
I meant this one: https://github.com/thesofproject/linux/pull/2172

Reply:
@ranj063 the RX mask was 0x3 for both amps, but the TX mask was different (0xC and 0x3) respectively. Same thing, you can extract data from common slots, you cannot drive the same slots.

Reply:
Ahh I see. makes sense. Thanks @plbossart

Reply:
> Hang on, there could be a problem here.
> The idea of using TDM slots is that when the slot configuration is the same, all amplifiers would receive the same data from the same slots and extract it
> For capture, you cannot use the same slots for the two amplifiers, that would lead to a bus clash.
> Can you clarify @RanderWang what setting the same TX mask for the amplifiers? How does this impact the bit allocation for the Source port of the amplifiers?

The DSM usage is: four channel capture stream and each amp generates two channels of IV. I first tried 0xc and 0x3, but only left channel feedback data was returned.  I struggled at here for a long time until I tested both 0x3. I got reply from @ryans-lee by email. please check his reply:

" I found ‘tx_mask’ is directly applied to 0x320 register which is Soundwire DP3_ChannelEn in ‘sdw_enable_disable_slave_ports’ function.
Each amp uses channel1 and channel2 for I and V respectively, so this value needs to be 3.
0x3(0y00000011) and 0xc(0y00001100) were for 8 slot TDM bitmask on I2S and this is no longer valid for Soundwire. **In Soundwire, slot allocation is done by 0x324, DP offsetctrl register. Offset configuration is done automatically in ‘sdw_compute_slave_ports'.** ‘stream_config.ch_count' in ‘max98373_sdw_dai_hw_params' is used to calculate the offset.
This was 4, so offset was calculated to 0x91. Now channel size become 2, so offset is changed to 0x61 which is good value. "

My idea is: the two amp will work on different DP offset based on ch_mask in sdw_compute_slave_ports, so no clash. Please correct me.